src/plugins/hetero/README.md

```diff
@@ -0,0 +1,144 @@
+# Inference Engine hetero plugin design overview
+
+## Subgraphs selection
+
+Algorithm:
+
+For each plugin
+1. Select *root* node
+    * Node not in subgraph previously constructed
+    * Affinity is equal to plugin name
+2. Select adjacent node to any node in already subgraph which is not in rejected list
+    * if there are no such nodes **end**
+3. Check selected node has same affinity
+4. Add node to subgraph if check was successful or add to rejected list otherwise
+5. Check global condition
+    * Nodes in rejected list can never be added to subgraph
+    * Nodes not in subgraph and not in rejected list can possibly be added later
+    * Check subgraph topology (the only check now is there are no indirect subgraph self-references)
+6. If global condition was failed remove last node from subgraph, add it to rejected list and go to step 5
+    * we can rollback multiple times here because rejected list is changed every time
+7. Go to step 2
+
+Example:
+```
+    1
+    |
+    2
+   / \
+  3   4
+   \ /
+    5
+    |
+    6
+    |
+    7
+```
```

Comment:
Could you use `mermaid` instead?

Reply:
Fixed [cb23e8c](https://github.com/openvinotoolkit/openvino/pull/15372/commits/cb23e8c2673c9e389719112fe8ad68c797f3aeb1)